compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -550,22 +553,80 @@ public void visit(BLangFunction funcNode) {
         funcNode.symbol = dupFuncSymbol;
         BInvokableType dupFuncType = (BInvokableType) dupFuncSymbol.type;
 
-        //write closure vars
-        funcNode.closureVarSymbols.stream()
-                .filter(symbol -> !isFunctionArgument(symbol, funcNode.symbol.params))
-                .forEach(symbol -> {
-                    symbol.closure = true;
-                    dupFuncSymbol.params.add(0, symbol);
-                    dupFuncType.paramTypes.add(0, symbol.type);
-                });
+        // Create a map symbol for the function statement
+        BUnionType constraintTypeOfMap = new BUnionType(null,
+                new LinkedHashSet<BType>() {{ add(symTable.anyType); add(symTable.errorType); }}, true);
+        BVarSymbol mapSymbol = new BVarSymbol(0, names.fromString("$mapFunc$" + funClosureMapCount),
```

Comment:
Shall we use string constants for the string literals? 
eg: "$mapFunc$", "$paramsMap$"